components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +332,74 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'USER' || typeValue === 'TEAM') {
+            this.allIdentity = typeValue;
+            this.updateValidations('identity', true);
+            return true;
+          } else {
+            this.updateValidations('identity', false);
+            this.expressionForm.get('identity').reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'TOKEN' || (identityValue && identityValue !== '*') ) {
+          this.updateValidations('name', true);
+          return true;
+        } else {
+          this.updateValidations('name', false);
+          this.expressionForm.get('name').reset();
+          return false;
+        }
+        break;
+      default:
+        return false;
+    }
+  }
+
+  private updateValidations(inputName: string, active: boolean): void {
+    const updateInput = this.expressionForm.get(inputName);
+
+    active === true
+      ? updateInput.setValidators([Validators.required,
+        Validators.pattern(Regex.patterns.NON_BLANK)])
+      : updateInput.clearValidators();
+
+      updateInput.updateValueAndValidity();
+    }
+
+
+  private displayExpressionOutput(formValues: object): void {
+    const entries = Object.entries(formValues);
```

Comment:
Add strong typing for entries
```suggestion
    const entries: string[][] = Object.entries(formValues);
```
... and that will give key and value strong types too.
Also add type for `output`.